core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -1150,6 +1187,24 @@ private ExprValue exprDayOfYear(ExprValue date) {
     return new ExprIntegerValue(date.dateValue().getDayOfYear());
   }
 
+  private ExprValue exprExtract(ExprValue part, ExprValue datetime) {
+    String partName = part.stringValue().toUpperCase();
+
+    if(partName == "QUARTER"){
+      return exprQuarter(datetime);
+    }
+
+    if (extract_formats.containsKey(partName)) {
```

Comment:
I think you can omit `if` - it is (well, should be) filtered on the parsing stage.

Reply:
Fixed in https://github.com/Bit-Quill/opensearch-project-sql/pull/221/commits/d15c26ba044d8db36dad2dcf197757404aeb8ece